hub/core/tests/test_tensor.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
+
+from hub.core.tensor import (
+    append_tensor,
+    create_tensor,
+    extend_tensor,
+    read_samples_from_tensor,
+)
+from hub.tests.common import TENSOR_KEY
+from hub.core.typing import StorageProvider
+from .common import parametrize_all_storages_and_caches
+
+
+@parametrize_all_storages_and_caches
+def test_fixed_shape(storage: StorageProvider):
+    create_tensor(TENSOR_KEY, storage, dtype="int64")
+
+    a1 = np.arange(100).reshape(50, 2)
+    a2 = np.arange(1000).reshape(10, 50, 2)
+    append_tensor(a1, TENSOR_KEY, storage)
+    extend_tensor(a2, TENSOR_KEY, storage)
+
+    out = read_samples_from_tensor(TENSOR_KEY, storage)
+    np.testing.assert_array_equal(out, [a1, *a2])
+
+
+@parametrize_all_storages_and_caches
+def test_dynamic_shape(storage: StorageProvider):
+    create_tensor(TENSOR_KEY, storage, dtype="int64")
+
+    a1 = np.arange(100).reshape(25, 4)
+    a2 = np.arange(1000).reshape(10, 50, 2)
+    a3 = np.arange(10).reshape(10, 1)
```

Comment:
Only cases I would add are leading 1s, to make sure append/extend are respecting the batch property

Reply:
done